### ISSUE TITLE
Documentation: Simple Typescript usage example

### ISSUE THREAD

**Issue by alaincao** [jdalrymple/gitbeaker#384]
**Description**
Heylow, I'd like to have a simple example on where I can find the types returned by the different functions. e.g.
```typescript
import * as gitlab from 'gitlab'
const aaa = new gitlab.Gitlab({ host:'foo', token:'bar' });
const bbb : WHAT_AM_I??? = aaa;
```
The thing is that I would like to pass them to e.g. a function, and for now I am using `any` where the `WHAT_AM_I???` is, but that kinda defeats the purpose ...  
I am hitting a wall for *days*, here ; I have a hint with the
```typescript
import * as APIServices from 'gitlab/dist/services';
```
I saw in issue # 257, but still, no idea how to use ...

**Proposal**
I'd love to find a simple example e.g. in the `Readme.md` file or somewhere easily reachable ...

**Thanks !**

**Comment by jdalrymple** [comment]
Ill throw something together in the next couple days!

**Comment by jdalrymple** [comment]
So I haven't been able to get around to this. If anyone has time this would be a good first issue!

**Comment by ksaaskil** [comment]
I was able to infer the type of `new Gitlab(...)` with following:
```ts
type ConstructedType<Constructor> = Constructor extends {
  new (...args: any[]): infer B;
}
  ? B
  : never;

type GitlabType = ConstructedType<typeof Gitlab>;
```
If that looks good, I can add a PR to add that in documentation and/or export the types for easier access?

**Comment by ksaaskil** [comment]
No wait I'm dumb, one can use the built-in `InstanceType`:
```ts
type GitlabApi = InstanceType<typeof Gitlab>;
```

**Comment by Bauke** [comment]
Although probably a rather big job, would it be possible to also get proper types for the responses to API calls?

For example calling `Projects.show()` returns a `Promise<GetResponse>`. If I then wanted to get the project ID or anything else that does in fact exist on the result, TypeScript errors out and says it doesn't exist on the `GetResponse` type (which would be correct):

```
source/edit-label.ts:24:48 - error TS2339: Property 'id' does not exist on type 'GetResponse'.
  Property 'id' does not exist on type 'object'.

24     log((await api.Projects.show(projectName)).id);
```

 Only by casting it to a type you made yourself with the right properties or `any` can you then access it:

```ts
    log((await api.Projects.show(projectName) as any).id);
    // Doesn't error and lets you access it
```

**Comment by jdalrymple** [comment]
@Bauke Eventually thats the plan, by pulling the information straight from gitlab's docs. #104 but until the expose proper api docs, its going to be a little difficult :(

**Comment by Telokis** [comment]
What is missing in order for us to have proper typings for `MergeRequests`?

**Comment by jdalrymple** [comment]
A chunk is already supported, but i need to finally merge some of the PR's. Im just waiting on the first release of the latest version of the library before doing so and THATS being held up by finishing the automatic release pipeline with semantic-release-npmx. Almost there, just been a crazy couple months :/

**Comment by Granjow** [comment]
For a while now all Typing support has gone and the types file lists that:

```typescript
export declare const Groups: any, GroupAccessRequests: any, GroupBadges: any, GroupCustomAttributes: any, GroupIssueBoards: any // etc.
```

Are the types (which are still supported in the original [gitlab](https://www.npmjs.com/package/gitlab) package) going to come back?

**Comment by jdalrymple** [comment]
They should be! Ill look into it, sorry about that :/

**Comment by jdalrymple** [comment]
@Granjow , continuing the conversation in #793

**Comment by simpsamu** [comment]
This is tremendously important .

**Comment by jdalrymple** [comment]
> For a while now all Typing support has gone and the types file lists that:
> 
> ```ts
> export declare const Groups: any, GroupAccessRequests: any, GroupBadges: any, GroupCustomAttributes: any, GroupIssueBoards: any // etc.
> ```
> 
> Are the types (which are still supported in the original [gitlab](https://www.npmjs.com/package/gitlab) package) going to come back?

This should be fixed now

**Comment by jdalrymple** [comment]
> What is missing in order for us to have proper typings for `MergeRequests`?

This have also been released

**Comment by jdalrymple** [comment]
Closing in favour of #492
